packages/util-ts/README.md

```diff
@@ -142,8 +146,99 @@ function validateUser(params: { email: string; phone: string }): ServiceResult<{
   return success({ id: "user-123" });
 }
 
-ok(E.isLeft(validateUser({ email: "invalidEmail", phone: "invalidPhoneNumber" })));
-ok(E.isRight(validateUser({ email: "user@example.com", phone: "555-555-5555" })));
+ok(isFailure(validateUser({ email: "invalidEmail", phone: "invalidPhoneNumber" })));
+ok(isSuccess(validateUser({ email: "user@example.com", phone: "555-555-5555" })));
+```
+
+</embedex>
+
+#### `tryCatchAsync`
+
+<embedex source="packages/util-ts/examples/tryCatchAsync.ts">
+
+```ts
+import { ok, strictEqual } from "node:assert/strict";
+
+import { isFailure, isSuccess, ServiceError, tryCatchAsync } from "@clipboard-health/util-ts";
+
+async function example() {
+  const successResult = await tryCatchAsync(
+    async () => {
+      const response = await fetch("https://jsonplaceholder.typicode.com/posts/1");
+      return (await response.json()) as { id: number };
+    },
+    (error) => new ServiceError(`Failed to fetch: ${String(error)}`),
+  );
+
+  ok(isSuccess(successResult));
+  strictEqual(successResult.right.id, 1);
+
+  const failureResult = await tryCatchAsync(
+    async () => await Promise.reject(new Error("Network error")),
+    (error) => new ServiceError(`Failed to fetch: ${String(error)}`),
+  );
+
+  ok(isFailure(failureResult));
+  strictEqual(failureResult.left.issues[0]?.message, "Failed to fetch: Error: Network error");
+}
+
+// eslint-disable-next-line unicorn/prefer-top-level-await
+void example();
+```
+
+</embedex>
+
+#### `tryCatch`
+
+<embedex source="packages/util-ts/examples/tryCatch.ts">
+
+```ts
+import { ok, strictEqual } from "node:assert/strict";
+
+import { isFailure, isSuccess, parseJson, ServiceError, tryCatch } from "@clipboard-health/util-ts";
+
+const successResult = tryCatch(
+  () => parseJson<{ name: string }>('{"name": "John"}'),
+  (error) => new ServiceError(`Parse error: ${String(error)}`),
+);
+
+ok(isSuccess(successResult));
+strictEqual(successResult.right.name, "John");
```

Comment:
Thanks for this PR. Would it be possible to "hide" the `right` in the value as well? So it could be used like `successResult.value.name` or even `successResult.name` (or `successResult().name` if it helps)?

Reply:
Have we no faith that our engineers can learn that left means error and right means success?

I created aliases assuming you use the `success` and `failure` helpers to construct `ServiceResult`:
- `right` -> `value`
- `left` -> `error`
- `isRight` -> `isSuccess`